### DISCUSSION TITLE
How to use the cooling modules?

### DISCUSSION THREAD

**Discussion by Rogerod23** [PrincetonUniversity/athena#632] [Q&A] [answered]
Hello all, I am new to Athena++. I intend to use the code to produce simulations of wind-blown bubbles. So far, I have been able to produce a simple 2D bubble with a constant mass-loss rate and constant density without cooling. Now, I want to use radiative cooling to start producing more realistic models, but I do not understand very well how to use the cooling and heating modules. 
According to the documentation and the chemical networks in the source code, there are several cooling and heating functions implemented into Athena (https://github.com/PrincetonUniversity/athena/wiki/Chemistry). However, I do not understand how can I implement them into my code. I have seen some of the problem generator files that make use of the chemical networks but its not clear to me how these are built into the code and operate for every time-step. 

Do I need to call the cooling function at the start of the problem? e.g. use it in the problem generator block as it is shown in the problem generator files. Or do I need to call the cooling function in the MeshBlock::UserWoorkInLoop as it is stated in the wiki in the problem generator section?

I also want to include a constant ionizing photon rate in the simulation in order to also produce the HII region structure of the bubble. I noticed that the chemistry module also includes a constant radiation source but I do not understand how this is also called in the code from the problem generator file. 

Thank you very much for your responses.

-Rogelio.

**Comment by ltlancas** [comment, marked as answer]
Hi Rogelio, sounds like an interesting problem! While I haven't used the chemistry module myself, I believe in order to set it up you simply need to compile using the instructions on the [Wiki](https://github.com/PrincetonUniversity/athena/wiki/Chemistry) with `./configure.py --prob=[problem] --chemistry=[network] --ode_solver=[ode_solver]` for your problem you would probably want

`./configure.py --prob=[your-problem-name] --chemistry=gow17 --ode_solver=cvode`

the cooling and heating should then be taken care of, as described in the Wiki and in Munan Gong's [methods paper](https://arxiv.org/abs/2305.04965), though you will have to make sure you compile with the correct amount of passive scalars and initialize the chemical species in `MeshBlock::ProblemGenerator()` in the problem generator file.

The chemistry module is currently only meant to handle a single radiation band. In particular it is set up to handle the interaction of Far Ultraviolet (FUV) light with interstellar chemistry and uses a simplified 6-ray approximation to attenuate the radiation field. If you wanted to include photo-ionizing radiation you would likely have to couple this to one of the other radiation transport methods like the [Non-Relativistic Radiation Transport](https://github.com/PrincetonUniversity/athena/wiki/Non-relativistic-Radiation-Transport) module. I'm not sure if anyone has done something like this yet working from the Athena++ master branch.

On another note, I am working on a very similar problem and would love to chat with you about your plans. - Lachlan

**Reply by Rogerod23** [reply]
Dear Lachlan, Thank you for your answer. I would be glad to meet at a later time. You can write to me at rorozco@astro.unam.mx.

-Rogelio.
